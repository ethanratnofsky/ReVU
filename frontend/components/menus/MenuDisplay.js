import {ScrollView, Text, SafeAreaView} from 'react-native';
import menuPageStyles from './MenuStyles';
import FoodList from './FoodList';
import { DINING_HALLS } from '../../constants';

const MENU = [
    {"time": "Breakfast", "items": ["Tater Tots", "Scrambeled Eggs", "Maple Syrup Infused Waffles", "Vegan Chipotle Eggs"]},
    {"time": "Lunch", "items": ["Glazed Pork", "Vegan Pot Pie", "Green Beans", "Potato Wedges"]},
    {"time": "Dinner", "items": ["Fried Okra", "Vegan Sloppy Joe", "Lemon Pepper Chicken Wings", "White Rice"]}
]


const MenuDisplay = ({navigation, route}) => {

    const { diningHallId } = route.params;
    const diningHallName = DINING_HALLS[diningHallId].name;
    // TODO: Fetch
    return (
        <SafeAreaView style={menuPageStyles.container}>
            <Text style={menuPageStyles.titleText}>{diningHallName}</Text>
            <ScrollView>
                {MENU.map((time, i) => {
                    return (
                        <FoodList key={i} timeOfDay={time["time"]} list={time["items"]}/>
                    )
                })}
            </ScrollView>
        </SafeAreaView>
    );

}

export default MenuDisplay;