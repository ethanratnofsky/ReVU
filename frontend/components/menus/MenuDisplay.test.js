import React from 'react';
const renderer = require('react-test-renderer');
import MenuDisplay from './MenuDisplay';

describe('<MenuDisplay />', () => {
  it('has 3 children', () => {
    const tree = renderer.create(<MenuDisplay />).toJSON();
    expect(tree.children.length).toBe(3);
  });
});

it('renders correctly', () => {
    const tree = renderer.create(<MenuDisplay />).toJSON();
    expect(tree).toMatchSnapshot();
  });