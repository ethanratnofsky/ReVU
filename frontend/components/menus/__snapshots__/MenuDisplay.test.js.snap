// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#1C1C1C",
      "height": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "position": "absolute",
        "top": 45,
        "width": "90%",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
          "padding": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontSize": 28,
            "fontWeight": "bold",
          }
        }
      >
        ‚Üê
      </Text>
    </View>
  </View>
  <Text
    style={
      Object {
        "color": "#FFFFFF",
        "fontFamily": "LibreCaslonText_700Bold",
        "fontSize": 20,
        "marginBottom": 15,
        "marginTop": 15,
      }
    }
  >
    Rand Dining Hall
  </Text>
  <RCTScrollView>
    <View>
      <View>
        <Text
          style={
            Object {
              "borderColor": "#CFAE70",
              "borderWidth": 3,
              "color": "#FFFFFF",
              "fontFamily": "LibreCaslonText_700Bold",
              "fontSize": 18,
              "marginBottom": 5,
              "marginLeft": 15,
              "marginRight": 15,
              "marginTop": 20,
              "padding": 5,
              "textAlign": "center",
              "width": 300,
            }
          }
        >
          Breakfast
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFFFFF",
              "borderColor": "#CFAE70",
              "borderWidth": 3,
              "height": 40,
              "justifyContent": "center",
              "marginBottom": 5,
              "marginLeft": 15,
              "marginRight": 15,
              "marginTop": 5,
              "padding": 5,
              "width": 300,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#1C1C1C",
                "fontFamily": "LibreCaslonText_400Regular",
              }
            }
          >
            Banana Cakes
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFFFFF",
              "borderColor": "#CFAE70",
              "borderWidth": 3,
              "height": 40,
              "justifyContent": "center",
              "marginBottom": 5,
              "marginLeft": 15,
              "marginRight": 15,
              "marginTop": 5,
              "padding": 5,
              "width": 300,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#1C1C1C",
                "fontFamily": "LibreCaslonText_400Regular",
              }
            }
          >
            Green Beans
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFFFFF",
              "borderColor": "#CFAE70",
              "borderWidth": 3,
              "height": 40,
              "justifyContent": "center",
              "marginBottom": 5,
              "marginLeft": 15,
              "marginRight": 15,
              "marginTop": 5,
              "padding": 5,
              "width": 300,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#1C1C1C",
                "fontFamily": "LibreCaslonText_400Regular",
              }
            }
          >
            Vegan Chili
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFFFFF",
              "borderColor": "#CFAE70",
              "borderWidth": 3,
              "height": 40,
              "justifyContent": "center",
              "marginBottom": 5,
              "marginLeft": 15,
              "marginRight": 15,
              "marginTop": 5,
              "padding": 5,
              "width": 300,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#1C1C1C",
                "fontFamily": "LibreCaslonText_400Regular",
              }
            }
          >
            Sloppy Joe
          </Text>
        </View>
      </View>
      <View>
        <Text
          style={
            Object {
              "borderColor": "#CFAE70",
              "borderWidth": 3,
              "color": "#FFFFFF",
              "fontFamily": "LibreCaslonText_700Bold",
              "fontSize": 18,
              "marginBottom": 5,
              "marginLeft": 15,
              "marginRight": 15,
              "marginTop": 20,
              "padding": 5,
              "textAlign": "center",
              "width": 300,
            }
          }
        >
          Breakfast
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFFFFF",
              "borderColor": "#CFAE70",
              "borderWidth": 3,
              "height": 40,
              "justifyContent": "center",
              "marginBottom": 5,
              "marginLeft": 15,
              "marginRight": 15,
              "marginTop": 5,
              "padding": 5,
              "width": 300,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#1C1C1C",
                "fontFamily": "LibreCaslonText_400Regular",
              }
            }
          >
            Banana Cakes
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFFFFF",
              "borderColor": "#CFAE70",
              "borderWidth": 3,
              "height": 40,
              "justifyContent": "center",
              "marginBottom": 5,
              "marginLeft": 15,
              "marginRight": 15,
              "marginTop": 5,
              "padding": 5,
              "width": 300,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#1C1C1C",
                "fontFamily": "LibreCaslonText_400Regular",
              }
            }
          >
            Green Beans
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFFFFF",
              "borderColor": "#CFAE70",
              "borderWidth": 3,
              "height": 40,
              "justifyContent": "center",
              "marginBottom": 5,
              "marginLeft": 15,
              "marginRight": 15,
              "marginTop": 5,
              "padding": 5,
              "width": 300,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#1C1C1C",
                "fontFamily": "LibreCaslonText_400Regular",
              }
            }
          >
            Vegan Chili
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFFFFF",
              "borderColor": "#CFAE70",
              "borderWidth": 3,
              "height": 40,
              "justifyContent": "center",
              "marginBottom": 5,
              "marginLeft": 15,
              "marginRight": 15,
              "marginTop": 5,
              "padding": 5,
              "width": 300,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#1C1C1C",
                "fontFamily": "LibreCaslonText_400Regular",
              }
            }
          >
            Sloppy Joe
          </Text>
        </View>
      </View>
      <View>
        <Text
          style={
            Object {
              "borderColor": "#CFAE70",
              "borderWidth": 3,
              "color": "#FFFFFF",
              "fontFamily": "LibreCaslonText_700Bold",
              "fontSize": 18,
              "marginBottom": 5,
              "marginLeft": 15,
              "marginRight": 15,
              "marginTop": 20,
              "padding": 5,
              "textAlign": "center",
              "width": 300,
            }
          }
        >
          Breakfast
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFFFFF",
              "borderColor": "#CFAE70",
              "borderWidth": 3,
              "height": 40,
              "justifyContent": "center",
              "marginBottom": 5,
              "marginLeft": 15,
              "marginRight": 15,
              "marginTop": 5,
              "padding": 5,
              "width": 300,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#1C1C1C",
                "fontFamily": "LibreCaslonText_400Regular",
              }
            }
          >
            Banana Cakes
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFFFFF",
              "borderColor": "#CFAE70",
              "borderWidth": 3,
              "height": 40,
              "justifyContent": "center",
              "marginBottom": 5,
              "marginLeft": 15,
              "marginRight": 15,
              "marginTop": 5,
              "padding": 5,
              "width": 300,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#1C1C1C",
                "fontFamily": "LibreCaslonText_400Regular",
              }
            }
          >
            Green Beans
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFFFFF",
              "borderColor": "#CFAE70",
              "borderWidth": 3,
              "height": 40,
              "justifyContent": "center",
              "marginBottom": 5,
              "marginLeft": 15,
              "marginRight": 15,
              "marginTop": 5,
              "padding": 5,
              "width": 300,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#1C1C1C",
                "fontFamily": "LibreCaslonText_400Regular",
              }
            }
          >
            Vegan Chili
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFFFFF",
              "borderColor": "#CFAE70",
              "borderWidth": 3,
              "height": 40,
              "justifyContent": "center",
              "marginBottom": 5,
              "marginLeft": 15,
              "marginRight": 15,
              "marginTop": 5,
              "padding": 5,
              "width": 300,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#1C1C1C",
                "fontFamily": "LibreCaslonText_400Regular",
              }
            }
          >
            Sloppy Joe
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
